samples/nrf9160/modem_shell/src/cloud/cloud_mqtt_shell.c

```diff
@@ -240,13 +241,21 @@ static void cmd_cloud_connect(const struct shell *shell, size_t argc, char **arg
 
 static void cmd_cloud_disconnect(const struct shell *shell, size_t argc, char **argv)
 {
-	int err = nrf_cloud_disconnect();
+	int err;
 
+	if (!initialized) {
+		mosh_print("Not connected to nRF Cloud");
+		return;
+	}
+
+	/* Stop possibly pending reconnection attempt. */
+	k_work_cancel_delayable(&cloud_reconnect_work);
+
+	err = nrf_cloud_disconnect();
```

Comment:
Not caused by this PR but disconnecting and connecting doesn't seem to open a valid connection:
```
cloud connect
cloud disconnect
cloud connect
```

The last connection will show but the connection doesn't work:
```
mosh:~$ cloud connect
Endpoint: mqtt.nrfcloud.com
nRF Cloud connection already established
[00:00:45.357,696] <dbg> nrf_cloud: start_connection_poll: Connection poll in progress
RRC mode: Idle
```

I guess this might be something we cannot control at all...

Reply:
That was NCSDK-19889, should be fixed now by #10342.

Reply:
Ok, this was related to the poll bug.